### PR TITLE
Feature GitHub auth

### PR DESCRIPTION
**Pull request by gwenf**
* Added github authentication flow and components
* User information is saved in Vuex store

### CODE REVIEW

File: src/router/index.js [hunk 1]
```diff
@@ -71,6 +72,23 @@ export const routes = [
...
+      // If the user is logged in
+      const token = localStorage.getItem('token');
+      if (token) {
+        next();
+      }
+
+      // If the user just registered with Github
+      const code = to.params.code;
+      if (code) {
+
```
Comment (TheJaredWilcurt) [inline src/router/index.js:86]: ???

File: src/router/index.js [hunk 2]
```diff
@@ -71,6 +72,23 @@ export const routes = [
...
+      }
+      // if not, then check if the user has a code & make a
+      // call to the back-end with code to get token
+      next();
+      // next('/login')
```
Comment (TheJaredWilcurt) [inline src/router/index.js]: `beforeEnter` needs some cleanup

File: src/router/index.js [hunk 3]
```diff
@@ -109,37 +132,37 @@ const router = new VueRouter({
...
-  }
-}
+// export function checkRoutePermissions (to, from, next) {
+//   if (!to.meta.groups.includes(store.state.currentUser.groups)) {
+//     store.commit('setLastURLRequested', to.path);
+//     next('/access-denied');
+//   } else {
+//     next();
+//   }
+// }
```
Comment (TheJaredWilcurt) [inline src/router/index.js]: If this comment block is not needed, it can be removed.

File: src/views/auth/Login.vue [hunk 4]
```diff
@@ -1,45 +1,21 @@
...
   },
   methods: {
-    ...mapActions(['login']),
     logUserIn: function () {
-      const payload = {
-        username: this.email,
-        password: this.password
-      };
-      this.login(payload);
+      window.location.href = 'https://github.com/login/oauth/authorize?client_id=b1566423c56583f16eb0';
```
Comment (TheJaredWilcurt) [inline src/views/auth/Login.vue:18]:
```suggestion
      // Routes to GitHub login page. Once authenticated, GitHub will route back to `/auth/callback?code=1234`
      window.location.href = 'https://github.com/login/oauth/authorize?client_id=b1566423c56583f16eb0';
```

File: package.json [hunk 5]
```diff
@@ -17,6 +17,7 @@
     "axios": "0.19.2",
     "foundation-sites": "6.6.3",
     "vue": "^2.6.11",
+    "vue-authenticate": "1.4.1",
```
Comment (TheJaredWilcurt) [inline package.json]: Where is this being used

File: src/router/index.js (hunk 3 above)
Comment (gwenf) [reply src/router/index.js]: I left this in since you added it. I thought maybe you wanted it for later.

File: package.json (hunk 5 above)
Comment (gwenf) [reply package.json]: nowhere. removing

File: src/router/index.js (hunk 2 above)
Comment (gwenf) [reply src/router/index.js]: I removed it.

File: src/router/index.js (hunk 3 above)
Comment (gwenf) [reply src/router/index.js]: removed it

File: src/store/index.js [hunk 6]
```diff
@@ -6,33 +6,55 @@ import router from '../router/index.js';
...
+};
+
 const API_URL = 'http://localhost:8000/api/v1';
 
 export default new Vuex.Store({
   state: {
     currentUser: {},
-    lastURLRequested: null
+    lastURLRequested: null,
+    isAuthenticated: false
```
Comment (TheJaredWilcurt) [inline src/store/index.js]:
```suggestion
    lastURLRequested: null
```
doesn't seem to be used anywhere. easy to add back in if needed, but will probably just use `currentUser` to check, unless I'm misunderstanding some of the auth/local storage set up where they could have a currentUser, but not be auth'd

File: src/store/index.js [hunk 7]
```diff
@@ -6,33 +6,55 @@ import router from '../router/index.js';
...
   },
   mutations: {
     setCurrentUser: function (state, user) {
       state.currentUser = user;
     },
     setLastURLRequested: function (state, url) {
       state.lastURLRequested = url;
+    },
+    isAuthenticated: function (state, payload) {
+      state.isAuthenticated = payload.isAuthenticated;
```
Comment (TheJaredWilcurt) [inline src/store/index.js]:
```suggestion
```

File: src/views/auth/Register.vue [hunk 8]
```diff
@@ -66,8 +68,14 @@ export default {
...
   methods: {
+    ...mapActions(['register']),
     registerUser: function () {
-
+      this.$store.dispatch('register', {
+        name: 'gwenf',
+        email: 'gwenfaraday@protonmail.com',
+        password: 'o6qSOp5eO7U8'
+      })
+      // this.$router.push('/github-auth');
```
Comment (TheJaredWilcurt) [inline src/views/auth/Register.vue]: this seems suspect

Comment (gwenf) [reply src/views/auth/Register.vue]: lol. Don't worry, it's not my password. I just tried it temporarily for something. I need to remove it.

**Review by TheJaredWilcurt — APPROVED** [review]

**Pull request merged**